File: api/generate.js
const https = require('https');
const crypto = require('crypto');

const CLOUD_NAME = 'dsicpafgz';
const API_KEY = '989282614168298';
const API_SECRET = 'hyP7yp3zKmfkj-g--cdQLTZr9iw';

function generateSVG(data) {
  const { name='', daily=0, surgeryFixed=0, inpatient=0, fracture=0, accident=0, accDeath=0, disability=0, critical=0, cancer=0, ltc=0, life=0 } = data;

  const W = 1600, H = 1680;
  const CX = 800, CY = 820;
  const R = 155;
  const G = 10;

  const TLW = CX - R - G;
  const TLH = CY - R - G;
  const TRW = W - CX - R - G;
  const BLH = H - CY - R - G - 60;

  function ok(v,s){ return Number(v)>=s; }

  function pill(v,s,unit){
    const isOk=ok(v,s);
    return {bg:isOk?'#bbf7d0':'#fecaca',color:isOk?'#14532d':'#b91c1c',text:`${isOk?'✓':'!'} ${Number(v).toLocaleString()} ${unit}`};
  }
  function pillB(has){
    return {bg:has?'#bbf7d0':'#fecaca',color:has?'#14532d':'#b91c1c',text:has?'✓ 已投保':'! 未投保'};
  }

  // Draw a row with icon square, label, pill
  function row(cx, cy, iconBg, iconColor, iconPath, label, p, cardW) {
    const pillW = 230;
    const pillX = cx + cardW - pillW - 20;
    return `
      <rect x="${cx+16}" y="${cy-22}" width="44" height="44" rx="10" fill="${iconBg}"/>
      <text x="${cx+38}" y="${cy+8}" font-size="26" text-anchor="middle" fill="${iconColor}" font-family="Arial,sans-serif" font-weight="bold">${iconPath}</text>
      <text x="${cx+72}" y="${cy+8}" font-size="30" fill="#1f2937" font-family="Arial,sans-serif">${label}</text>
      <rect x="${pillX}" y="${cy-18}" width="${pillW}" height="36" rx="18" fill="${p.bg}"/>
      <text x="${pillX+pillW/2}" y="${cy+8}" font-size="22" fill="${p.color}" text-anchor="middle" font-weight="bold" font-family="Arial,sans-serif">${p.text}</text>`;
  }

  const lifeOk = ok(life,500);

  function header(x, y, w, hBg, iconBg, iconColor, iconText, title, titleColor, subtitle, subColor) {
    return `
      <rect x="${x}" y="${y}" width="${w}" height="100" rx="24" fill="${hBg}"/>
      <rect x="${x+16}" y="${y+16}" width="68" height="68" rx="16" fill="${iconBg}"/>
      <text x="${x+50}" y="${y+60}" font-size="38" text-anchor="middle" fill="${iconColor}" font-family="Arial,sans-serif" font-weight="bold">${iconText}</text>
      <text x="${x+100}" y="${y+52}" font-size="44" fill="${titleColor}" font-weight="bold" font-family="Arial,sans-serif">${title}</text>
      <text x="${x+100}" y="${y+80}" font-size="24" fill="${subColor}" font-family="Arial,sans-serif">${subtitle}</text>`;
  }

  return `<?xml version="1.0" encoding="UTF-8"?>
<svg width="${W}" height="${H}" xmlns="http://www.w3.org/2000/svg">
<defs>
  <linearGradient id="bg" x1="0" y1="0" x2="1" y2="1">
    <stop offset="0%" stop-color="#eef2ff"/>
    <stop offset="100%" stop-color="#fdf4ff"/>
  </linearGradient>
  <linearGradient id="ring" x1="0" y1="0" x2="1" y2="1">
    <stop offset="0%" stop-color="#8b5cf6"/>
    <stop offset="100%" stop-color="#ec4899"/>
  </linearGradient>
  <filter id="sh"><feDropShadow dx="0" dy="4" stdDeviation="14" flood-color="#0000001a"/></filter>
</defs>
<rect width="${W}" height="${H}" fill="url(#bg)"/>

<!-- TL 小疾病 -->
<rect x="0" y="0" width="${TLW}" height="${TLH}" rx="28" fill="white" filter="url(#sh)"/>
${header(0,0,TLW,'#eff6ff','#dbeafe','#1d4ed8','＋','小疾病','#1d4ed8','住院相關保障','#93c5fd')}
<line x1="16" y1="104" x2="${TLW-16}" y2="104" stroke="#e0f2fe" stroke-width="1.5"/>
${row(0, 150, '#dbeafe', '#1d4ed8', '床', '住院日額（實支+定額）', pill(daily,3000,'元/天'), TLW)}
${row(0, 202, '#dbeafe', '#1d4ed8', '✂', '手術（實支實付）', pill(surgeryFixed,50000,'元'), TLW)}
${row(0, 254, '#dbeafe', '#1d4ed8', '刀', '手術（定額手術）', pill(surgeryFixed,50000,'元'), TLW)}
${row(0, 306, '#dbeafe', '#1d4ed8', '藥', '雜費', pill(inpatient,100000,'元'), TLW)}

<!-- TR 小意外 -->
<rect x="${CX+R+G}" y="0" width="${TRW}" height="${TLH}" rx="28" fill="white" filter="url(#sh)"/>
${header(CX+R+G,0,TRW,'#fff1f2','#ffe4e6','#be123c','＋','小意外','#be123c','意外相關保障','#fda4af')}
<line x1="${CX+R+G+16}" y1="104" x2="${CX+R+G+TRW-16}" y2="104" stroke="#ffe4e6" stroke-width="1.5"/>
${row(CX+R+G, 150, '#ffe4e6', '#be123c', '急', '意外門診', pill(accident,10000,'元'), TRW)}
${row(CX+R+G, 202, '#ffe4e6', '#be123c', '骨', '骨折未住院', pill(fracture,30000,'元'), TRW)}
${row(CX+R+G, 254, '#ffe4e6', '#be123c', '院', '意外住院日額', pillB(Number(daily)>0), TRW)}

<!-- BL 大疾病 -->
<rect x="0" y="${CY+R+G}" width="${TLW}" height="${BLH}" rx="28" fill="white" filter="url(#sh)"/>
${header(0,CY+R+G,TLW,'#f5f3ff','#ede9fe','#6d28d9','癌','大疾病','#6d28d9','重大疾病保障','#c4b5fd')}
<line x1="16" y1="${CY+R+G+104}" x2="${TLW-16}" y2="${CY+R+G+104}" stroke="#ede9fe" stroke-width="1.5"/>
${row(0, CY+R+G+150, '#ede9fe', '#6d28d9', '心', '重大傷病', pill(critical,100,'萬'), TLW)}
${row(0, CY+R+G+202, '#ede9fe', '#6d28d9', '盾', '重大疾病', pill(critical,100,'萬'), TLW)}
${row(0, CY+R+G+254, '#ede9fe', '#6d28d9', '症', '癌症一次金', pill(cancer,100,'萬'), TLW)}
${row(0, CY+R+G+306, '#ede9fe', '#6d28d9', '療', '化療/放療', pill(0,1,'萬'), TLW)}
${row(0, CY+R+G+358, '#ede9fe', '#6d28d9', '院', '癌症住院', pill(0,3000,'元/日'), TLW)}
${row(0, CY+R+G+410, '#ede9fe', '#6d28d9', '照', '長照月給付', pill(ltc,30000,'元/月'), TLW)}

<!-- BR 大意外 -->
<rect x="${CX+R+G}" y="${CY+R+G}" width="${TRW}" height="${BLH}" rx="28" fill="white" filter="url(#sh)"/>
${header(CX+R+G,CY+R+G,TRW,'#fff7ed','#ffedd5','#c2410c','⚡','大意外','#c2410c','意外相關保障','#fdba74')}
<line x1="${CX+R+G+16}" y1="${CY+R+G+104}" x2="${CX+R+G+TRW-16}" y2="${CY+R+G+104}" stroke="#ffedd5" stroke-width="1.5"/>
${row(CX+R+G, CY+R+G+150, '#ffedd5', '#c2410c', '故', '意外身故', pill(accDeath,500,'萬'), TRW)}
${row(CX+R+G, CY+R+G+202, '#ffedd5', '#c2410c', '殘', '殘廢（1-11級）', pill(disability,500,'萬'), TRW)}
${row(CX+R+G, CY+R+G+254, '#ffedd5', '#c2410c', '全', '全殘', pill(disability,500,'萬'), TRW)}

<!-- Cross lines -->
<line x1="${CX}" y1="${TLH}" x2="${CX}" y2="${CY-R-G}" stroke="#d1d5db" stroke-width="2" stroke-dasharray="10,7"/>
<line x1="${CX}" y1="${CY+R+G}" x2="${CX}" y2="${CY+R+G+BLH}" stroke="#d1d5db" stroke-width="2" stroke-dasharray="10,7"/>
<line x1="${TLW}" y1="${CY}" x2="${CX-R-G}" y2="${CY}" stroke="#d1d5db" stroke-width="2" stroke-dasharray="10,7"/>
<line x1="${CX+R+G}" y1="${CY}" x2="${W}" y2="${CY}" stroke="#d1d5db" stroke-width="2" stroke-dasharray="10,7"/>

<!-- Center -->
<circle cx="${CX}" cy="${CY}" r="${R+12}" fill="white" filter="url(#sh)"/>
<circle cx="${CX}" cy="${CY}" r="${R+4}" fill="white"/>
<circle cx="${CX}" cy="${CY}" r="${R+4}" fill="none" stroke="url(#ring)" stroke-width="10"/>
<!-- Shield path -->
<g transform="translate(${CX-40},${CY-90})">
  <path d="M40 0 L80 15 L80 50 C80 70 60 85 40 90 C20 85 0 70 0 50 L0 15 Z" fill="#8b5cf6" opacity="0.15"/>
  <path d="M40 8 L72 20 L72 50 C72 66 56 78 40 82 C24 78 8 66 8 50 L8 20 Z" fill="none" stroke="#8b5cf6" stroke-width="3"/>
  <path d="M28 45 L36 53 L54 35" fill="none" stroke="#8b5cf6" stroke-width="5" stroke-linecap="round"/>
</g>
<text x="${CX}" y="${CY+16}" font-size="40" fill="#7c3aed" text-anchor="middle" font-weight="bold" font-family="Arial,sans-serif">壽險</text>
<text x="${CX}" y="${CY+58}" font-size="${Number(life)>999?'28':'34'}" fill="${lifeOk?'#14532d':'#b91c1c'}" text-anchor="middle" font-weight="bold" font-family="Arial,sans-serif">${life?life+'萬':'未投保'}</text>
${!lifeOk?`<text x="${CX}" y="${CY+84}" font-size="18" fill="#b91c1c" text-anchor="middle" font-family="Arial,sans-serif">建議500萬以上</text>`:`<text x="${CX}" y="${CY+84}" font-size="18" fill="#14532d" text-anchor="middle" font-family="Arial,sans-serif">✓ 保額充足</text>`}

<!-- Footer -->
<text x="${CX}" y="${H-24}" font-size="28" fill="#7c3aed" text-anchor="middle" font-weight="bold" font-family="Arial,sans-serif">安心守護・全面保障</text>
<text x="${CX}" y="${H-2}" font-size="20" fill="#9ca3af" text-anchor="middle" font-family="Arial,sans-serif">${name}・保寶險保障分析報告</text>
</svg>`;
}

module.exports = async (req, res) => {
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'POST, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');
  if (req.method === 'OPTIONS') return res.status(200).end();
  if (req.method !== 'POST') return res.status(405).json({ error: 'Method not allowed' });
  try {
    const data = req.body;
    const svg = generateSVG(data);
    const svgB64 = Buffer.from(svg).toString('base64');
    const boundary = 'B' + Date.now();
    const timestamp = Math.floor(Date.now() / 1000);
    const sig = crypto.createHash('sha1').update(`format=png&timestamp=${timestamp}${API_SECRET}`).digest('hex');
    const uploadBody = [
      `--${boundary}\r\nContent-Disposition: form-data; name="file"\r\n\r\ndata:image/svg+xml;base64,${svgB64}`,
      `--${boundary}\r\nContent-Disposition: form-data; name="format"\r\n\r\npng`,
      `--${boundary}\r\nContent-Disposition: form-data; name="api_key"\r\n\r\n${API_KEY}`,
      `--${boundary}\r\nContent-Disposition: form-data; name="timestamp"\r\n\r\n${timestamp}`,
      `--${boundary}\r\nContent-Disposition: form-data; name="signature"\r\n\r\n${sig}`,
      `--${boundary}--`
    ].join('\r\n');
    const url = await new Promise((resolve, reject) => {
      const r = https.request({
        hostname: 'api.cloudinary.com',
        path: `/v1_1/${CLOUD_NAME}/image/upload`,
        method: 'POST',
        headers: { 'Content-Type': `multipart/form-data; boundary=${boundary}`, 'Content-Length': Buffer.byteLength(uploadBody) }
      }, res2 => {
        let d=''; res2.on('data',c=>d+=c);
        res2.on('end',()=>{ try{ const r=JSON.parse(d); if(r.secure_url) resolve(r.secure_url.replace(/\.[^.]+$/,'.png')); else reject(new Error(d)); }catch(e){reject(new Error(d));} });
      });
      r.on('error',reject); r.write(uploadBody); r.end();
    });
    res.json({ url });
  } catch(e) {
    console.error('Error:', e.message);
    res.status(500).json({ error: e.message });
  }
};

File: api/api/generate.py
from PIL import Image, ImageDraw, ImageFont
import json
import base64
import hashlib
import hmac
import time
import urllib.request
import urllib.parse
import io
import os

CLOUD_NAME = 'dsicpafgz'
API_KEY = '989282614168298'
API_SECRET = 'hyP7yp3zKmfkj-g--cdQLTZr9iw'

def upload_to_cloudinary(img_bytes):
    timestamp = str(int(time.time()))
    sig_str = f"format=png&timestamp={timestamp}{API_SECRET}"
    signature = hashlib.sha1(sig_str.encode()).hexdigest()
    
    b64 = base64.b64encode(img_bytes).decode()
    data_uri = f"data:image/png;base64,{b64}"
    
    import urllib.request
    import urllib.parse
    
    boundary = f"Boundary{int(time.time())}"
    parts = []
    
    def add_field(name, value):
        parts.append(f'--{boundary}\r\nContent-Disposition: form-data; name="{name}"\r\n\r\n{value}')
    
    add_field('file', data_uri)
    add_field('format', 'png')
    add_field('api_key', API_KEY)
    add_field('timestamp', timestamp)
    add_field('signature', signature)
    parts.append(f'--{boundary}--')
    
    body = '\r\n'.join(parts).encode('utf-8')
    
    req = urllib.request.Request(
        f'https://api.cloudinary.com/v1_1/{CLOUD_NAME}/image/upload',
        data=body,
        headers={'Content-Type': f'multipart/form-data; boundary={boundary}'}
    )
    
    with urllib.request.urlopen(req) as resp:
        result = json.loads(resp.read())
        url = result.get('secure_url', '')
        if url and not url.endswith('.png'):
            url = url.rsplit('.', 1)[0] + '.png'
        return url

def score_pct(v, s):
    v, s = float(v), float(s)
    if s == 0: return 100
    pct = min(100, int(v / s * 100))
    return pct

def draw_rounded_rect(draw, xy, radius, fill=None, outline=None, width=1):
    x1, y1, x2, y2 = xy
    draw.rounded_rectangle([x1, y1, x2, y2], radius=radius, fill=fill, outline=outline, width=width)

def generate_image(data):
    name = data.get('name', '')
    daily = float(data.get('daily', 0))
    surgery = float(data.get('surgeryFixed', 0))
    inpatient = float(data.get('inpatient', 0))
    fracture = float(data.get('fracture', 0))
    accident = float(data.get('accident', 0))
    acc_death = float(data.get('accDeath', 0))
    disability = float(data.get('disability', 0))
    critical = float(data.get('critical', 0))
    cancer = float(data.get('cancer', 0))
    ltc = float(data.get('ltc', 0))
    life = float(data.get('life', 0))

    # Colors
    BG = (245, 243, 255)
    WHITE = (255, 255, 255)
    BLUE = (29, 78, 216)
    BLUE_LIGHT = (239, 246, 255)
    BLUE_MID = (191, 219, 254)
    RED = (190, 18, 60)
    RED_LIGHT = (255, 241, 242)
    RED_MID = (253, 164, 175)
    PURPLE = (109, 40, 217)
    PURPLE_LIGHT = (245, 243, 255)
    PURPLE_MID = (196, 181, 253)
    ORANGE = (194, 65, 12)
    ORANGE_LIGHT = (255, 247, 237)
    ORANGE_MID = (253, 186, 116)
    GREEN = (20, 83, 45)
    GREEN_BG = (187, 247, 208)
    DARK_RED = (185, 28, 28)
    RED_BG = (254, 202, 202)
    GRAY = (156, 163, 175)
    DARK = (31, 41, 55)
    TITLE_BG = (124, 58, 237)

    W, H = 1600, 1750
    img = Image.new('RGB', (W, H), BG)
    draw = ImageDraw.Draw(img)

    # Try to load font
    try:
        font_path = '/usr/share/fonts/truetype/dejavu/DejaVuSans.ttf'
        font_bold_path = '/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf'
        f_sm = ImageFont.truetype(font_path, 28)
        f_md = ImageFont.truetype(font_path, 34)
        f_lg = ImageFont.truetype(font_bold_path, 44)
        f_xl = ImageFont.truetype(font_bold_path, 60)
        f_xxl = ImageFont.truetype(font_bold_path, 80)
        f_title = ImageFont.truetype(font_bold_path, 52)
        f_xs = ImageFont.truetype(font_path, 22)
    except:
        f_sm = f_md = f_lg = f_xl = f_xxl = f_title = f_xs = ImageFont.load_default()

    # Header
    draw_rounded_rect(draw, [0, 0, W, 110], 0, fill=TITLE_BG)
    draw.text((W//2, 55), '保單健診分析報告', font=f_title, fill=WHITE, anchor='mm')

    # Layout
    CX, CY = 800, 830
    R = 200
    G = 12
    TLW = CX - R - G
    TRW = W - CX - R - G
    TOP_H = CY - R - G - 110
    BOT_H = H - CY - R - G - 120

    def score_color(pct):
        if pct >= 80: return GREEN, GREEN_BG
        if pct >= 50: return ORANGE, ORANGE_LIGHT
        return DARK_RED, RED_BG

    def pill(draw, cx, cy, val, std, unit, w=220, h=36):
        pct = score_pct(val, std)
        isOk = val >= std
        bg = GREEN_BG if isOk else RED_BG
        color = GREEN if isOk else DARK_RED
        icon = '✓' if isOk else '!'
        text = f"{icon} {int(val):,} {unit}"
        x1, y1 = cx - w//2, cy - h//2
        draw_rounded_rect(draw, [x1, y1, x1+w, y1+h], h//2, fill=bg)
        draw.text((cx, cy), text, font=f_xs, fill=color, anchor='mm')

    def pill_bool(draw, cx, cy, has, w=180, h=36):
        bg = GREEN_BG if has else RED_BG
        color = GREEN if has else DARK_RED
        text = '✓ 已投保' if has else '! 未投保'
        x1, y1 = cx - w//2, cy - h//2
        draw_rounded_rect(draw, [x1, y1, x1+w, y1+h], h//2, fill=bg)
        draw.text((cx, cy), text, font=f_xs, fill=color, anchor='mm')

    def progress_bar(draw, x, y, w, pct, color):
        draw_rounded_rect(draw, [x, y, x+w, y+14], 7, fill=(229, 231, 235))
        bar_w = max(14, int(w * pct / 100))
        draw_rounded_rect(draw, [x, y, x+bar_w, y+14], 7, fill=color)

    def draw_card(x, y, w, h, header_bg, border_color, icon_bg, icon_char, icon_color, title, title_color, subtitle, sub_color, rows, pct, pct_label):
        # Card background
        draw_rounded_rect(draw, [x, y, x+w, y+h], 24, fill=WHITE, outline=border_color, width=2)
        # Header
        draw_rounded_rect(draw, [x, y, x+w, y+100], 24, fill=header_bg)
        draw.rectangle([x, y+76, x+w, y+100], fill=header_bg)
        # Icon
        draw_rounded_rect(draw, [x+16, y+16, x+76, y+76], 14, fill=icon_bg)
        draw.text((x+46, y+46), icon_char, font=f_lg, fill=icon_color, anchor='mm')
        # Title
        draw.text((x+94, y+46), title, font=f_lg, fill=title_color, anchor='lm')
        draw.text((x+94, y+76), subtitle, font=f_xs, fill=sub_color, anchor='lm')
        # Separator
        draw.line([x+16, y+100, x+w-16, y+100], fill=(229,231,235), width=2)
        # Rows
        for i, (label, func) in enumerate(rows):
            ry = y + 140 + i * 54
            draw.text((x+20, ry), label, font=f_sm, fill=DARK, anchor='lm')
            func(x+w-130, ry)
        # Progress
        bar_y = y + h - 52
        bar_color = pct[1]
        draw.text((x+16, bar_y - 2), pct_label, font=f_xs, fill=pct[0], anchor='lm')
        progress_bar(draw, x+16, bar_y+22, w-32, pct[2], bar_color)

    # Scores
    tl_pct = int((score_pct(daily,3000)*0.4 + score_pct(surgery,50000)*0.3 + score_pct(inpatient,100000)*0.3))
    tr_pct = int((score_pct(accident,10000)*0.4 + score_pct(fracture,30000)*0.3 + (100 if daily>0 else 0)*0.3))
    bl_pct = int((score_pct(critical,100)*0.25 + score_pct(cancer,100)*0.25 + score_pct(ltc,30000)*0.25 + 25))
    br_pct = int((score_pct(acc_death,500)*0.4 + score_pct(disability,500)*0.4 + (100 if disability>0 else 0)*0.2))
    total_score = int((tl_pct*0.3 + tr_pct*0.2 + bl_pct*0.3 + br_pct*0.2))

    sc_tl = score_color(tl_pct)
    sc_tr = score_color(tr_pct)
    sc_bl = score_color(bl_pct)
    sc_br = score_color(br_pct)

    pill_cx_tl = TLW - 130
    pill_cx_tr = CX + R + G + TRW - 130

    # TL Card
    draw_card(
        0, 110, TLW, TOP_H,
        BLUE_LIGHT, BLUE_MID, (219,234,254), '＋', BLUE,
        '小疾病保障', BLUE, '住院相關保障', (147,197,253),
        [
            ('住院日額（實支+定額）', lambda cx,cy: pill(draw,cx,cy,daily,3000,'元/天')),
            ('手術（實支實付）', lambda cx,cy: pill(draw,cx,cy,surgery,50000,'元')),
            ('手術（定額手術）', lambda cx,cy: pill(draw,cx,cy,surgery,50000,'元')),
            ('雜費', lambda cx,cy: pill(draw,cx,cy,inpatient,100000,'元')),
        ],
        (sc_tl[0], sc_tl[1], tl_pct),
        f'保障充足度 {tl_pct}%'
    )

    # TR Card
    draw_card(
        CX+R+G, 110, TRW, TOP_H,
        RED_LIGHT, RED_MID, (255,228,230), '＋', RED,
        '小意外保障', RED, '意外相關保障', (253,164,175),
        [
            ('意外門診', lambda cx,cy: pill(draw,cx,cy,accident,10000,'元')),
            ('骨折未住院', lambda cx,cy: pill(draw,cx,cy,fracture,30000,'元')),
            ('意外住院日額', lambda cx,cy: pill_bool(draw,cx,cy,daily>0)),
        ],
        (sc_tr[0], sc_tr[1], tr_pct),
        f'保障充足度 {tr_pct}%'
    )

    # BL Card
    draw_card(
        0, CY+R+G, TLW, BOT_H,
        PURPLE_LIGHT, PURPLE_MID, (237,233,254), '癌', PURPLE,
        '大疾病保障', PURPLE, '重大疾病保障', (196,181,253),
        [
            ('重大傷病', lambda cx,cy: pill(draw,cx,cy,critical,100,'萬')),
            ('重大疾病', lambda cx,cy: pill(draw,cx,cy,critical,100,'萬')),
            ('癌症一次金', lambda cx,cy: pill(draw,cx,cy,cancer,100,'萬')),
            ('化療/放療', lambda cx,cy: pill(draw,cx,cy,0,1,'萬')),
            ('癌症住院', lambda cx,cy: pill(draw,cx,cy,0,3000,'元/日')),
            ('長照月給付', lambda cx,cy: pill(draw,cx,cy,ltc,30000,'元/月')),
        ],
        (sc_bl[0], sc_bl[1], bl_pct),
        f'保障充足度 {bl_pct}%'
    )

    # BR Card
    draw_card(
        CX+R+G, CY+R+G, TRW, BOT_H,
        ORANGE_LIGHT, ORANGE_MID, (255,237,213), '⚡', ORANGE,
        '大意外保障', ORANGE, '意外相關保障', (253,186,116),
        [
            ('意外身故', lambda cx,cy: pill(draw,cx,cy,acc_death,500,'萬')),
            ('殘廢（1-11級）', lambda cx,cy: pill(draw,cx,cy,disability,500,'萬')),
            ('全殘', lambda cx,cy: pill(draw,cx,cy,disability,500,'萬')),
        ],
        (sc_br[0], sc_br[1], br_pct),
        f'保障充足度 {br_pct}%'
    )

    # Center circle
    circle_img = Image.new('RGBA', (W, H), (0,0,0,0))
    circle_draw = ImageDraw.Draw(circle_img)
    circle_draw.ellipse([CX-R-20, CY-R-20, CX+R+20, CY+R+20], fill=(220,200,255,80))
    circle_draw.ellipse([CX-R-8, CY-R-8, CX+R+8, CY+R+8], fill=(255,255,255,255), outline=(139,92,246), width=12)
    img = Image.alpha_composite(img.convert('RGBA'), circle_img).convert('RGB')
    draw = ImageDraw.Draw(img)

    # Center circle - 壽險保額
    draw.text((CX, CY-80), '壽險保額', font=f_lg, fill=PURPLE, anchor='mm')
    life_text = f"{int(life)}萬" if life > 0 else '未投保'
    life_color = GREEN if life >= 500 else DARK_RED
    draw.text((CX, CY), life_text, font=f_xxl, fill=life_color, anchor='mm')
    status = '✓ 保額充足' if life >= 500 else '建議500萬以上'
    status_color = GREEN if life >= 500 else DARK_RED
    draw.text((CX, CY+80), status, font=f_md, fill=status_color, anchor='mm')

    # Suggestions section
    issues = []
    if accident < 10000: issues.append('建議加強小意外\n門診保障額度')
    if critical < 100 and cancer < 100: issues.append('可考慮提升重大\n疾病保障額度')
    if ltc < 30000: issues.append('建議規劃長照\n月給付保障')
    if not issues:
        issues = ['整體保障良好，\n持續定期檢視']

    sug_y = H - 200
    draw_rounded_rect(draw, [16, sug_y, W-16, H-60], 20, fill=WHITE, outline=PURPLE_MID, width=2)
    draw.text((36, sug_y+20), '醫療保障建議', font=f_md, fill=PURPLE, anchor='lm')

    col_w = (W - 32) // max(len(issues), 1)
    icons = ['🛡', '＋', '✓']
    for i, tip in enumerate(issues[:3]):
        sx = 32 + i * col_w + col_w//2
        sy = sug_y + 65
        draw_rounded_rect(draw, [sx-col_w//2+10, sug_y+44, sx+col_w//2-10, H-76], 14, fill=PURPLE_LIGHT)
        draw.text((sx-col_w//2+26, sy+2), icons[i % len(icons)], font=f_md, fill=PURPLE, anchor='lm')
        for j, line in enumerate(tip.split('\n')):
            draw.text((sx, sy+10+j*32), line, font=f_xs, fill=DARK, anchor='mm')

    draw.text((W//2, H-30), f'本分析僅供參考・實際保障內容以保單條款為準', font=f_xs, fill=GRAY, anchor='mm')

    buf = io.BytesIO()
    img.save(buf, format='PNG', dpi=(144,144))
    return buf.getvalue()

def handler(request):
    if request.method == 'OPTIONS':
        return {'statusCode': 200, 'headers': {'Access-Control-Allow-Origin': '*', 'Access-Control-Allow-Methods': 'POST,OPTIONS', 'Access-Control-Allow-Headers': 'Content-Type'}, 'body': ''}
    
    if request.method != 'POST':
        return {'statusCode': 405, 'body': json.dumps({'error': 'Method not allowed'})}
    
    try:
        data = json.loads(request.body)
        img_bytes = generate_image(data)
        url = upload_to_cloudinary(img_bytes)
        return {
            'statusCode': 200,
            'headers': {'Access-Control-Allow-Origin': '*', 'Content-Type': 'application/json'},
            'body': json.dumps({'url': url})
        }
    except Exception as e:
        import traceback
        return {
            'statusCode': 500,
            'headers': {'Access-Control-Allow-Origin': '*', 'Content-Type': 'application/json'},
            'body': json.dumps({'error': str(e), 'trace': traceback.format_exc()})
        }
